libs/source/Socket.cpp

```diff
@@ -23,7 +23,7 @@ Socket::Socket(Socket::type stype) {
 
   if ((SocketFileDescriptor = socket(AF_INET, type, proto)) == -1) {
     LOG(IPC, Sev::Error, "socket() failed");
-    exit(1);
+    throw std::runtime_error("system error - socket() failed");
```

Comment:
Wouldn't it be useful to log the error code or include it in the throw message in this case?

Reply:
Not in this case. Either a valid socket fd is returned or -1